core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4176,23 +4217,54 @@ private static void computeVersion(ServletContext context) {
     }
 
     /**
+     * The version number before it is "computed" (by a call to computeVersion()).
+     * @since FIXME
+     */
+    public static final String UNCOMPUTED_VERSION = "?";
+
+    /**
      * Version number of this Jenkins.
      */
-    public static String VERSION="?";
+    public static String VERSION = UNCOMPUTED_VERSION;
 
     /**
      * Parses {@link #VERSION} into {@link VersionNumber}, or null if it's not parseable as a version number
      * (such as when Jenkins is run with "mvn hudson-dev:run")
      */
-    public static VersionNumber getVersion() {
+    public @CheckForNull static VersionNumber getVersion() {
+        return toVersion(VERSION);
+    }
+
+    /**
+     * Get the stored version of Jenkins, as stored by
+     * {@link #doConfigSubmit(org.kohsuke.stapler.StaplerRequest, org.kohsuke.stapler.StaplerResponse)}.
+     * <p>
+     * Parses the version into {@link VersionNumber}, or null if it's not parseable as a version number
+     * (such as when Jenkins is run with "mvn hudson-dev:run")
+     * @since FIXME
+     */
+    public @CheckForNull static VersionNumber getStoredVersion() {
```

Comment:
again - the `public` `static` nature of this is nothing new and nothing to do with this PR.